### PR TITLE
updated core versions

### PR DESCRIPTION
**Pull request by fehguy**
Updated swagger-core, swagger-inflector to current releases

### CODE REVIEW

**Comment by fehguy** [comment]
@wing328 any ideas on this? I'll be able to look more shortly

**Comment by wing328** [comment]
Taking a look...

**Comment by wing328** [comment]
There's an issue with Travis CI at the moment:
> We're experiencing a partial API outage, which is affecting logs availability and our web performance for public repositories. 
Please refer to traviscistatus.com for more details.

and I couldn't get any details of the job (e.g log)

Ref: https://www.traviscistatus.com/incidents/hx7cnxbch9xf

**Comment by fehguy** [comment]
thanks @wing328.  I must have gotten lucky with the parser update

**Comment by fehguy** [comment]
I think there was a problem with the online petstore.  Restarting the builds...

**Comment by wing328** [comment]
The build failed with the following error:
```
[ERROR] /home/travis/build/swagger-api/swagger-codegen/samples/server/petstore/jaxrs/jersey2/src/gen/java/io/swagger/api/JacksonJsonProvider.java:[6,42] error: package com.fasterxml.jackson.datatype.joda does not exist
[ERROR] /home/travis/build/swagger-api/swagger-codegen/samples/server/petstore/jaxrs/jersey2/src/gen/java/io/swagger/api/JacksonJsonProvider.java:[7,39] error: package com.fasterxml.jackson.jaxrs.json does not exist
[ERROR] /home/travis/build/swagger-api/swagger-codegen/samples/server/petstore/jaxrs/jersey2/src/gen/java/io/swagger/api/JacksonJsonProvider.java:[15,41] error: cannot find symbol
[ERROR]   symbol: class JacksonJaxbJsonProvider
/home/travis/build/swagger-api/swagger-codegen/samples/server/petstore/jaxrs/jersey2/src/gen/java/io/swagger/api/JacksonJsonProvider.java:[22,32] error: cannot find symbol
[ERROR]   symbol:   class JodaModule
  location: class JacksonJsonProvider
/home/travis/build/swagger-api/swagger-codegen/samples/server/petstore/jaxrs/jersey2/src/gen/java/io/swagger/api/JacksonJsonProvider.java:[25,8] error: cannot find symbol
[INFO] 5 errors 
```
Ref: https://api.travis-ci.org/jobs/206653633/log.txt?deansi=true

I'm looking into the issue...

File: samples/server/petstore/jaxrs/jersey2/pom.xml [hunk 1]
```diff
@@ -163,7 +163,7 @@
     <java.version>1.7</java.version>
     <maven.compiler.source>${java.version}</maven.compiler.source>
     <maven.compiler.target>${java.version}</maven.compiler.target>
-    <swagger-core-version>1.5.9</swagger-core-version>
+    <swagger-core-version>1.5.12</swagger-core-version>
```
Comment (wing328) [inline samples/server/petstore/jaxrs/jersey2/pom.xml]: @fehguy This change results in the following errors:
```
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-compiler-plugin:3.1:compile (default-compile) on project swagger-jaxrs-server: Compilation failure: Compilation failure:
[ERROR] /private/tmp/master/swagger-codegen/samples/server/petstore/jaxrs/jersey2/src/gen/java/io/swagger/api/JacksonJsonProvider.java:[6,43] package com.fasterxml.jackson.datatype.joda does not exist
[ERROR] /private/tmp/master/swagger-codegen/samples/server/petstore/jaxrs/jersey2/src/gen/java/io/swagger/api/JacksonJsonProvider.java:[7,40] package com.fasterxml.jackson.jaxrs.json does not exist
[ERROR] /private/tmp/master/swagger-codegen/samples/server/petstore/jaxrs/jersey2/src/gen/java/io/swagger/api/JacksonJsonProvider.java:[15,42] cannot find symbol
[ERROR] symbol: class JacksonJaxbJsonProvider
[ERROR] /private/tmp/master/swagger-codegen/samples/server/petstore/jaxrs/jersey2/src/gen/java/io/swagger/api/JacksonJsonProvider.java:[22,33] cannot find symbol
[ERROR] symbol:   class JodaModule
[ERROR] location: class io.swagger.api.JacksonJsonProvider
[ERROR] /private/tmp/master/swagger-codegen/samples/server/petstore/jaxrs/jersey2/src/gen/java/io/swagger/api/JacksonJsonProvider.java:[25,9] cannot find symbol
[ERROR] symbol:   method setMapper(com.fasterxml.jackson.databind.ObjectMapper)
[ERROR] location: class io.swagger.api.JacksonJsonProvider
[ERROR] -> [Help 1]
```

Comment (wing328) [inline samples/server/petstore/jaxrs/jersey2/pom.xml]: If I manually add the following to the pom.xml, `mvn test` does not throw any error:
```
<!-- https://mvnrepository.com/artifact/com.fasterxml.jackson.core/jackson-core -->
<dependency>
    <groupId>com.fasterxml.jackson.core</groupId>
    <artifactId>jackson-core</artifactId>
    <version>2.8.7</version>
</dependency>
```

Comment (wing328) [inline samples/server/petstore/jaxrs/jersey2/pom.xml]: I've added back the dependency via https://github.com/swagger-api/swagger-codegen/pull/4881/commits/f15267c717f56803c518c118270c85c0ab473252

**Comment by fehguy** [comment]
@wing328 Found another missing one.  We'll see if it passes

**Comment by wing328** [comment]
@fehguy yup, I added those via https://github.com/swagger-api/swagger-codegen/pull/4881/commits/9e40700478a1da3c4df6ef713cca6cf9259e7774

**Comment by wing328** [comment]
The Travis job passed: https://travis-ci.org/swagger-api/swagger-codegen/builds/206707978

**Comment by fehguy** [comment]
@wing328 shippable is waiting (40 minutes now).  Unless you object, I'm going to merge anyway

**Comment by wing328** [comment]
No objection. Please go ahead.

(The Shippable test aims to catch issue with mustache templates (e.g. missing close tab)).

**Pull request merged**